XLS-0040d-decentralized-identity/README.md

```diff
@@ -0,0 +1,416 @@
+In this section, we describe the DID method specification that conforms to the requirements specified in the DID specification currently published by the W3C Credentials Community Group. For more information about DIDs and DID method specifications, refer to the DID [Primer](https://github.com/WebOfTrustInfo/rwot5-boston/blob/master/topics-and-advance-readings/did-primer.md). 
+
+### 4.2.1. XRP Ledger DID Scheme
+The DID format as defined by W3C is as follows:
+```
+"did:" method-name ":" method-specific-idstring
+```
+We propose the following format for DIDs on XRPL: 
+```
+"did:xrpl": network-id: xrpl-specific-idstring
+```
+
+The components specific to the XRPL network are the following:
+
+- `method-name`: The `"xrpl"` namestring specifies that this is a DID for XRP Ledger.
+A DID that uses this method MUST begin with the following prefix: `did:xrpl`. Per the DID specification, this string MUST be in lowercase. The remainder of the DID, after the prefix, is specified below.
+
+- `method-specific-idstring` is formed by `network-id` and `xrpl-specific-idstring`
+ 
+  - `network-id`: `network-id` is a chain ID which is an identifier of XRP ledger networks. It specifies the underlying network instance where the `DID` is stored. Per [XLS-37d specification](https://github.com/XRPLF/rippled/pull/4418), in XRPL Protocol Chains the Network ID should match the chosen peer port.  
+    
+ -  `xrpl-specific-idstring` is generated as described in the next section.
+
+    
+### 4.2.2. XRPL-specific-idstring Generation Method
+XRPL DID MUST be unique within the XRPL network. 
+
+`xrpl-specific-idstring` is the `AccountID` or the hex of `master public key` of the `DID` object's account. See [this](https://xrpl.org/accounts.html#address-encoding) for more details on XRPL public keys.
+
+Example:
+A valid DID for an XRPL network may be
+```
+did:xrpl:1:rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh
+```
+or
+```
+did:xrpl:1:0330E7FC9D56BB25D6893BA3F317AE5BCF33B3291BD63DB32654A313222F7FD020
+```
+The above DIDs are case-insensitive and will resolve to the same `DID` object.
+
+# 5. CRUD Operations
+In this section we outline the following four CRUD operations for ``did:xrpl`` method. 
+- `Create`
+- `Read`
+- `Update` 
+- `Delete` 
+
+## 5.1. DIDSet Transaction 
+The proposal introduces a new transaction type: **`DIDSet`** that can be used to:
+- `Create` a new `DID` object,
+- `Update` the existing `DID` object.   
+
+### 5.1.1 Example **`DIDSet`** JSON
+```
+{
+  "TransactionType": "DIDSet",
+  "Account": "rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh",
+  "Fee": "10",
+  "Sequence": 391,
+  "URI": "ipfs://bafybeigdyrzt5sfp7udm7hu76uh7y26nf4dfuylqabf3oclgtqy55fbzdi",
+  "Data": ,
+  "SigningPubKey":"0330E7FC9D56BB25D6893BA3F317AE5BCF33B3291BD63DB32654A313222F7FD020",
+  ...
+
+  }
+```
+## Create
+If the `DID` object associated with the `Account` does not exist, 
+  - If both `URI` and `Data` fields are not present in the transaction, then transaction fails.
```

Comment:
Should we also add `DIDDocument` here to align with https://github.com/XRPLF/XRPL-Standards/pull/136/files#diff-b0de935ca4c9ab9568454d8b2c3bd0d2637314702ece03008631d5f4f96e0bd3R104